invenio_app_ils/ui/src/invenio_app_ils/common/state/PatronDocumentRequests/actions.js

```diff
@@ -0,0 +1,48 @@
+import { IS_LOADING, SUCCESS, HAS_ERROR } from './types';
+import { documentRequest as documentRequestApi } from '../../api';
+import { sendErrorNotification } from '../../components/Notifications';
+import { ES_DELAY } from '../../config';
+
+const selectQuery = (patronPid, page = 1) => {
+  return documentRequestApi
+    .query()
+    .withPatronPid(patronPid)
+    .withPage(page)
+    .sortByNewest()
+    .qs();
+};
+
+export const fetchPatronDocumentRequests = (patronPid, page) => {
+  const fetchRequests = (patronPid, dispatch) => {
+    documentRequestApi
+      .list(selectQuery(patronPid, page))
+      .then(response => {
```

Comment:
In the future, keep in mind that we decided to use `await` instead :)

Reply:
Fixed!